.github/workflows/cli.yml

```diff
@@ -50,3 +50,34 @@ jobs:
           azdev test apic-extension --live
         env:
           USERASSIGNED_IDENTITY: ${{ secrets.USERASSIGNED_IDENTITY }}
+
+  notify:
+    if: failure()
+    runs-on: ubuntu-latest
+    needs: apic-extension-live-test
+    environment: engineering
+    steps:
+      - name: Generate Email Content
+        id: generate-email-content
+        run: |
+          echo "to=frankqian@microsoft.com,chyuan@microsoft.com,shenwe@microsoft.com" >> $GITHUB_ENV
+          echo "body=Workflow URL: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}" >> $GITHUB_ENV
+          echo "subject=[APICENTER] Weekly APIC Extension Live Test Failed" >> $GITHUB_ENV
+
+      - name: Send Email Notification
+        run: |
+          response=$(curl \
+            --request POST \
+            --header "Content-Type: application/x-www-form-urlencoded" \
+            --data "grant_type=client_credentials&client_id=${{ secrets.MAIL_CLIENT_ID }}&client_secret=${{ secrets.MAIL_CLIENT_SECRET }}&resource=https://management.core.windows.net" \
```

Comment:
can we not have client secret here? Maybe use federated credential or other approach for authentication?

Reply:
The security guideline is avoid using client secret, so let's try to follow the guideline if we can.